src/server.cpp

```diff
@@ -2557,6 +2557,13 @@ bool Server::addMediaFile(const std::string &filename,
 		return false;
 	}
 
+	const char *deprecated_ext[] = { ".bmp", nullptr };
+	if (!removeStringEnd(filename, deprecated_ext).empty())
```

Comment:
I appreciate the warning and the foresight here, but isn't this a bit too complex for the current use case?

Readability counts – and AFAIK all remaining media formats are both in use and not easy to replace.

Reply:
What exactly is your suggestion here?

Reply:
Couldn't this wait until 6.0.0 as a breaking change?

Reply:
> What exactly is your suggestion here?

I would probably have written something like this – I am not a C++ expert, so I have no idea anyone finds it more readable:

```
bool is_bmp(std::string const &text) {
  if ( text.length() < 4 ) return false;
  return ( 0 == text.compare( text.length() - 4, 4, ".bmp" ) );
}
```

In any case, wrapping it into a function makes it easy to unit test functionality.

Thinking about this a bit,  I would suggest to *also* place a warning in the decoder, to catch dynamic use – because that is the one thing that can not be easily fixed by just converting image files from one format to the other. Some mods have put TGA bitmaps in node or item meta … so who knows if anyone has ever done the same with BMP?

Reply:
So you're suggesting I replace 2 LOC that call a function that is already known to work with 5 LOC for a new function that was written from scratch, which we should apparently also unit test?

> I would suggest to also place a warning in the decoder, to catch dynamic use

This is already covered because dynamic media also goes through this code.